### ISSUE TITLE
Subresource Integrity vs no scripts

### ISSUE THREAD

**Issue by keganlance** [mozilla/http-observatory#389]
Currently having Subresource Integrity gives you 5 bonus points. These bonus points are not given when scripts are blocked alltogether with csp script-src: 'none'.

One would expect blocking all scripts would be more secure than having subresource integrity. So at least 5 bonus points should be given here.

**Comment by floatingatoll** [comment]
Scores beyond 100/100 with no penalties are considered “extra credit”, and
due to various circumstances such as the scoring variance you’ve noted it’s
impractical (or impossible!) for every site to reach a theoretical
“maximum” high score. The disparity between SRI and script-src: none has
been discussed a few times previously and afaik no changes were planned as
of the most recent one.

**Comment by keganlance** [comment]
Alright. Thanks for the info! Currently the observatory doesn't recognise integrity of link tags. Is this something planned?

**Comment by floatingatoll** [comment]
Could you provide a sample of either the HTML content you’re describing or
a URL where it can been seen and scanned for testing?

**Comment by keganlance** [comment]
<link rel="stylesheet" href="main.css?v=3" crossorigin="anonymous" integrity="sha384-TkigcQyhTBNKZaDfB6OAn9d1cFvVEYWE5duhTxMi3hRmLd8XHusJMpR1U05IZONh">

in document head. The integrity is correct.

**Comment by Malvoz** [comment]
The initial concern seems like a dupe to https://github.com/mozilla/http-observatory/issues/273, although this issue seems to have taken another direction.

**Comment by mozdevcontrib** [comment]
> `<link rel="stylesheet" href="main.css?v=3" crossorigin="anonymous" integrity="sha384-TkigcQyhTBNKZaDfB6OAn9d1cFvVEYWE5duhTxMi3hRmLd8XHusJMpR1U05IZONh">`
> 
> in document head. The integrity is correct.

Indeed, the Observatory stopped looking at style when taking SRI into account, no matter the fact that a malicious stylesheet injection can be as dangerous as a malicious script injection.
